agent/src/main/java/io/jenkins/plugins/remotingkafka/Agent.java

```diff
@@ -0,0 +1,78 @@
+package io.jenkins.plugins.remotingkafka;
+
+import io.jenkins.plugins.remotingkafka.kafkaclient.KafkaConsumerClient;
+import io.jenkins.plugins.remotingkafka.kafkaclient.KafkaProducerClient;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class Agent {
+    private static final Logger LOGGER = Logger.getLogger(Agent.class.getName());
+
+    private final Options options;
+
+    public Agent(Options options) {
+        this.options = options;
+        LOGGER.finest("Agent created with " + options);
+    }
+
+    public static void main(String... args) throws InterruptedException, IOException {
+        String arguments = Arrays.toString(args);
+        arguments = arguments.replaceAll("\n", "");
+        arguments = arguments.replaceAll("\r", "");
+        arguments = arguments.replaceAll(",", "");
+        LOGGER.info("Agent.main invoked with: " + arguments);
+
+        Options options = new Options();
+        Agent agent = new Agent(options);
+
+        CmdLineParser p = new CmdLineParser(options);
+        try {
+            p.parseArgument(args);
+        } catch (CmdLineException e) {
+            LOGGER.log(Level.SEVERE, "CmdLineException occurred during parseArgument", e);
+            p.printUsage(System.out);
+            System.exit(-1);
+        }
+
+        if (options.help) {
+            p.printUsage(System.out);
+            System.exit(0);
+        }
+
+        if (options.name == null) {
+            try {
+                agent.options.name = InetAddress.getLocalHost().getCanonicalHostName();
+            } catch (IOException e) {
+                LOGGER.severe("Failed to lookup the canonical hostname of this slave, please check system settings.");
+                LOGGER.severe("If not possible to resolve please specify a node name using the '-name' option");
+                System.exit(-1);
+            }
+        }
+
+        // Kafka setup.
+        KafkaProducerClient producer = KafkaProducerClient.getInstance();
+        producer.initProducer(options.kafkaURL);
+        KafkaConsumerClient consumer = KafkaConsumerClient.getInstance();
+        consumer.initConsumer(options.kafkaURL);
+        URL url = new URL(options.master);
+        System.out.println(url.getHost());
```

Comment:
It may make sense ti add some text to clarify what is the output. I would also suggest using `LOGGER` instead of printing to STDOUT